File: server/internal/testHelpers/testGuesses.go
package testHelpers

import "spelldle.com/server/shared/types"

const GameSessionID types.GameSessionID = "TEST"

// ROUND 1: All incorrect
// ROUND 2: Some correct
// ROUND 3: All correct
var TestGuesses = [3]types.GuessCategories{
	{
		Components:  []uint{10, 11},
		Class:       []uint{12, 13, 14},
		Effects:     []uint{15, 16, 17},
		Level:       []uint{0, 0},
		School:      9,
		CastingTime: 10,
		Range:       11,
		Target:      12,
		Duration:    13,
	},
	{
		Components:  []uint{1, 3, 4},
		Class:       []uint{3, 4, 10},
		Effects:     []uint{7, 10, 13, 14},
		Level:       []uint{1, 0},
		School:      0,
		CastingTime: 10,
		Range:       11,
		Target:      12,
		Duration:    13,
	},
	{
		Components:  []uint{1, 2},
		Class:       []uint{3, 4, 5},
		Effects:     []uint{6, 7, 8, 9},
		Level:       []uint{1, 1},
		School:      0,
		CastingTime: 1,
		Range:       2,
		Target:      3,
		Duration:    4,
	},
}

var TestResults = [3]types.GuessResults{
	{
		Components:  0,
		Class:       0,
		Effects:     0,
		Level:       0,
		School:      0,
		CastingTime: 0,
		Range:       0,
		Target:      0,
		Duration:    0,
	},
	{
		Components:  1,
		Class:       1,
		Effects:     1,
		Level:       1,
		School:      2,
		CastingTime: 0,
		Range:       0,
		Target:      0,
		Duration:    0,
	},
	{
		Components:  2,
		Class:       2,
		Effects:     2,
		Level:       2,
		School:      2,
		CastingTime: 2,
		Range:       2,
		Target:      2,
		Duration:    2,
	},
}

File: client/src/components/DEBUG/GuessInfoButton/TEST_GUESSES.ts
import { T_GUESS_CATEGORIES } from "../../../methods/guesses";

export const TEST_GUESS_CORRECT: T_GUESS_CATEGORIES = {
	school: 0,
	level: { level: 1, is_ritual: false },
	casting_time: 1,
	range: 2,
	target: 3,
	duration: 4,
	components: [1, 2],
	class: [3, 4, 5],
	effects: [6, 7, 8, 9],
};

export const TEST_GUESS_INCORRECT: T_GUESS_CATEGORIES = {
	school: 1,
	level: { level: 0, is_ritual: true },
	casting_time: 2,
	range: 4,
	target: 5,
	duration: 6,
	components: [3, 4],
	class: [7, 8, 9],
	effects: [1, 2, 3, 4],
};

export const TEST_GUESS_SOME_CORRECT: T_GUESS_CATEGORIES = {
	school: 1,
	level: { level: 1, is_ritual: true },
	casting_time: 1,
	range: 2,
	target: 3,
	duration: 4,
	components: [1, 3],
	class: [3, 6, 7],
	effects: [6, 7, 1, 2],
};
